ci/actions/test.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+set -ex
+
+cargo build --verbose
+cargo test --verbose -- --test-threads=1
+
+for example_dir in examples/*; do
+    if [ $example != 'examples/linalg' ]; then
+        pushd $example_dir
+        tox -c "tox.ini" -e py
```

Comment:
Thanks @mtreinish ! Maybe just,
```
pip install .
pytest
```
there? I'm not sure what's the use-case for tox, since each CI job will run tests only for one python version. Also the tox.ini itself looks quite outdated e.g. https://github.com/PyO3/rust-numpy/blob/master/examples/linalg/tox.ini

Reply:
I don't remember correctly, but maybe here `tox` is for avoiding `pip install -e .` in the local environment.

Reply:
I don't remember correctly, but maybe here `tox` is for avoiding `pip install -e .` in the local environment.

Reply:
The typical use case for tox in my mind is to have a reproducible and isolated environment for testing locally that will mirror the CI environment. So you can run `tox` locally and will run with the same environment that ci does. In the pure ci case I agree it doesn't add much typically (there is some benefit to isolating from system python with a venv as there can sometimes be weird cross interactions with system python packages and what your testing, but that shouldn't come into play for this usecase).